### DISCUSSION TITLE
Compiler not opensource? just STDLIB?

### DISCUSSION THREAD

**Discussion by code959437957** [modular/modular#2036] [General]
I have not seen your compiler code.

Am i missing ?

So you called you opensource ?

You know compiler is the kernel and the most important thing of the programming language?
right?

**Comment by rarebreed** [comment]
I was wondering when someone was going to make a comment like this.

Modular has not claimed they have opensourced the entirety of Mojo.  If you read their blog _carefully_, you will notice the following:

>  We have been progressively open-sourcing more of Mojo and parts of the MAX platform

Notice the words, _have been progressively_.  Since you may not be a native English speaker, this is the present perfect continuous tense, meaning that something has happened, and is _continuing_ to happen.  In other words, the verb is not the past tense.  It is not yet done or completed.  This means Modular is progressing to open source even more _in the future_.

> We released Mojo very early and have been driving steady improvements since May of 2023

As above, still in the present perfect continuous tense, means they are continuing to drive more improvements.

> We have been using this approach for our existing open source code, and will continue to release more code with the same approach.

Well, that's pretty obvious that Modular is saying they will be releasing more open source in the future.

> Early stages of our open source journey
The Mojo standard library is under heavy development and changing rapidly, so we've started by open sourcing its core modules. As such, we consider this to be an important starting point, not an end to our open source journey. We’ll similarly evolve and iterate our development and contribution processes as we learn together with the community. We'll be opening up a lot more code over time, including more of Mojo and parts of the broader MAX platform.

Again, crystal clear that modular is not done open sourcing everything and this is their first big starting point.

I wish Modular the best of luck communicating with the open community.

**Reply by code959437957** [reply]
Thanks for tips out. now i am sure i am not miss the important part of the Mojo.it not release compiler code now. but for next future.

**Comment by Sharktheone** [comment]
Open sourcing the stdlib is the first step of open sourcing the whole thing. The compiler will also be open sourced at some point. This is the same way LLVM, CLang and others have done it.

See https://docs.modular.com/mojo/faq#open-source for more info

**Comment by rarebreed** [comment]
Another thing to keep in mind is that the compiler (as I understand it) has a lot of MLIR in it.  I would imagine this to be a very niche subset of engineers capable of doing this.  I think the compiler won't become widely tinkered with until they can bootstrap the compiler in mojo itself.

https://mlir.llvm.org/docs/Tutorials/
https://mlir.llvm.org/docs/LangRef/

**Comment by code959437957** [comment]
Though i got information i need , the question it is about ask if we lost the release of mojo compiler source code. 
Looks it not really release it . so the thread looks complete as it need.
